client/ayon_core/pipeline/context_tools.py

```diff
@@ -579,3 +586,47 @@ def get_process_id():
     if _process_id is None:
         _process_id = str(uuid.uuid4())
     return _process_id
+
+
+def version_up_current_workfile():
+    """Function to increment and save workfile
+    """
+    host = registered_host()
+    if not host.has_unsaved_changes():
+        print("No unsaved changes, skipping file save..")
+        return
+
+    project_name = get_current_project_name()
+    folder_path = get_current_folder_path()
+    task_name = get_current_task_name()
+    host_name = get_current_host_name()
+
+    template_key = get_workfile_template_key_from_context(
+        project_name,
+        folder_path,
+        task_name,
+        host_name,
+    )
+    anatomy = Anatomy(project_name)
+
+    data = get_template_data_with_names(
+        project_name, folder_path, task_name, host_name
+    )
+    data["root"] = anatomy.roots
+
+    work_template = anatomy.get_template_item("work", template_key)
+
+    # Define saving file extension
+    extensions = host.get_workfile_extensions()
+    current_file = host.get_current_workfile()
+    if current_file:
+        extensions = [os.path.splitext(current_file)[-1]]
+
+    work_root = work_template["directory"].format_strict(data)
+    file_template = work_template["file"].template
+    workfile_path = get_last_workfile(
+        work_root, file_template, data, extensions, True
+    )
+    if os.path.exists(workfile_path):
```

Comment:
As said before - this check does not fix things. See [here](https://github.com/ynput/ayon-core/pull/545#discussion_r1627782264) - it's redundant code.

However, the issue presented [here](https://github.com/ynput/ayon-core/pull/545#discussion_r1627697790) I assume is still present?

Reply:
is there any way you can suggest fixing it?